.github/workflows/test.yaml

```diff
@@ -128,212 +129,84 @@ jobs:
     # secrets context is only available at this level.
     #
     # ref: https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - name: dbless
+          test: dbless
+        - name: postgres
+          test: postgres
+        - name: feature-gates
+          test: dbless
+          feature_gates: "GatewayAlpha=true,CombinedRoutes=false"
+        - name: dbless-knative
+          test: dbless.knative
+        - name: postgres-knative
+          test: postgres.knative
+        - name: enterprise-postgres
+          test: enterprise.postgres
+          enterprise: true
+
     steps:
 
     - uses: Kong/kong-license@master
-      if: env.PULP_PASSWORD != ''
+      if: env.PULP_PASSWORD != '' && matrix.enterprise
       id: license
       with:
         password: ${{ env.PULP_PASSWORD }}
 
     - name: checkout repository
-      if: env.PULP_PASSWORD != ''
+      if: ${{ (env.PULP_PASSWORD != '' && matrix.enterprise) || (!matrix.enterprise) }}
```

Comment:
Could we have a separate step that would calculate this condition and set it in a named env var so it's easier to understand what it means? :)

Reply:
Oh, ok. Makes sense 👍